ncm-download/src/main/perl/download.pm

```diff
@@ -20,6 +61,126 @@ $NCM::Component::download::NoActionSupported = 1;
 my $_gss_tmpdir;
 my $_cached_gss;
 
+# TODO: Move this together with the BEGIN{} elsewhere, e.g. CAF::Download or CCM::Fetch::Download
+
+=pod
+
+=head1 FUNCTIONS
+
+=over
+
+=item _lwp_ua
+
+Initialise LWP::UserAgent and run C<method> with arrayref C<args>.
+Best-effort to handle ssl setup, Net::SSL vs IO::Socket::SSL
+and verify_hostname.
+
+Returns the result of the method or undef.
+
+Options
+
+=over
+
+=item cacert: the CA file
+
+=item cadir: the CA path
+
+=item cert: the client certificate filename
+
+=item key: the client certificate private key filename
+
+=item gss_ccache: the C<KRB5CCNAME> environment variable
+
+=item timeout: set timeout (if defined)
+
+=back
+
+=cut
+
+
+sub _lwp_ua
+{
+    my ($self, $method, $args, %opts) = @_;
+
+    # This is a mess.
+
+    # Set this again; very old Net::HTTPS (like in EL5) does not set the class
+    # on the initial import in the BEGIN{} section
+    $ENV{PERL_NET_HTTPS_SSL_SOCKET_CLASS} = $_default_https_class;
```

Comment:
Perhaps these should be `local` maybe? So that other incantations and aliases are not affected by this environment variable?

I know, it would be silly to have different classes and CAs for different downloading portions, but just in case...